File: doctors_hub/update_frontend.py
import os
import re

src_dir = '/home/ltl/Tomal/project_doctors_hub/doctors_hub/src'
mock_data_path = os.path.join(src_dir, 'data', 'mockData.js')
constants_path = os.path.join(src_dir, 'data', 'constants.js')

if os.path.exists(mock_data_path):
    with open(mock_data_path, 'r', encoding='utf-8') as f:
        content = f.read()

    city_thanas_match = re.search(r'export const CITY_THANAS = \{[\s\S]*?\};', content)
    locations_match = re.search(r'export const LOCATIONS = \[[\s\S]*?\];', content)

    if city_thanas_match and locations_match:
        with open(constants_path, 'w', encoding='utf-8') as f:
            f.write(city_thanas_match.group(0) + '\n\n' + locations_match.group(0) + '\n')
        print('Created constants.js')

def replace_in_file(file_path):
    if not os.path.exists(file_path):
        return
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    original = content

    def replacer(match):
        imports = [i.strip() for i in match.group(1).split(',')]
        keep = [i for i in imports if i in ('LOCATIONS', 'CITY_THANAS')]
        if keep:
            return f"import {{ {', '.join(keep)} }} from '{match.group(2)}constants';"
        return ''

    content = re.sub(r"import\s+\{([^}]+)\}\s+from\s+'([^']*)mockData';", replacer, content)

    if content != original:
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)
        print('Updated ' + file_path)

files_to_update = [
    'components/Footer.jsx',
    'components/TopUtilityStrip.jsx',
    'views/AdminDashboard/components/modals/DiagnosticModal.jsx',
    'views/AdminDashboard/components/modals/HospitalModal.jsx',
    'views/AdminDashboard/context/AdminContext.jsx',
    'views/DiagnosticsSearch/DiagnosticsSearchPage.jsx',
    'views/DoctorSearch/DoctorSearchPage.jsx',
    'views/Home/HomePage.jsx',
    'views/Home/components/DiagnosticsSection.jsx',
    'views/Home/components/DoctorMonitorGrid.jsx',
    'views/Home/components/SpecialtyGrid.jsx',
    'views/Home/components/ThreeWayEngine.jsx',
    'views/HospitalDetail/HospitalDetailPage.jsx',
    'views/Hospitals/HospitalsPage.jsx'
]

for rel_path in files_to_update:
    abs_path = os.path.join(src_dir, rel_path)
    replace_in_file(abs_path)

if os.path.exists(mock_data_path):
    os.remove(mock_data_path)
    print('Deleted mockData.js')

File: doctors_hub/update_frontend.js
const fs = require('fs');
const path = require('path');

const srcDir = '/home/ltl/Tomal/project_doctors_hub/doctors_hub/src';
const mockDataPath = path.join(srcDir, 'data', 'mockData.js');
const constantsPath = path.join(srcDir, 'data', 'constants.js');

if (!fs.existsSync(mockDataPath)) {
  console.log('mockData.js already removed.');
} else {
  const content = fs.readFileSync(mockDataPath, 'utf8');
  
  // Extract LOCATIONS and CITY_THANAS
  const cityThanasMatch = content.match(/export const CITY_THANAS = \{[\s\S]*?\};/);
  const locationsMatch = content.match(/export const LOCATIONS = \[[\s\S]*?\];/);
  
  if (cityThanasMatch && locationsMatch) {
    fs.writeFileSync(constantsPath, cityThanasMatch[0] + '\n\n' + locationsMatch[0] + '\n');
    console.log('Created constants.js');
  }
}

// Function to replace strings in files
function replaceInFile(filePath, replacements) {
  if (!fs.existsSync(filePath)) return;
  let fileContent = fs.readFileSync(filePath, 'utf8');
  let original = fileContent;
  for (const { regex, replacement } of replacements) {
    fileContent = fileContent.replace(regex, replacement);
  }
  if (fileContent !== original) {
    fs.writeFileSync(filePath, fileContent);
    console.log('Updated ' + filePath);
  }
}

const filesToUpdate = [
  'components/Footer.jsx',
  'components/TopUtilityStrip.jsx',
  'views/AdminDashboard/components/modals/DiagnosticModal.jsx',
  'views/AdminDashboard/components/modals/HospitalModal.jsx',
  'views/AdminDashboard/context/AdminContext.jsx',
  'views/DiagnosticsSearch/DiagnosticsSearchPage.jsx',
  'views/DoctorSearch/DoctorSearchPage.jsx',
  'views/Home/HomePage.jsx',
  'views/Home/components/DiagnosticsSection.jsx',
  'views/Home/components/DoctorMonitorGrid.jsx',
  'views/Home/components/SpecialtyGrid.jsx',
  'views/Home/components/ThreeWayEngine.jsx',
  'views/HospitalDetail/HospitalDetailPage.jsx',
  'views/Hospitals/HospitalsPage.jsx'
];

for (const relPath of filesToUpdate) {
  const absPath = path.join(srcDir, relPath);
  replaceInFile(absPath, [
    // Replace imports from mockData
    { regex: /import \{([^}]*)\} from '([^']*)mockData';/g, replacement: (match, p1, p2) => {
      // Check if LOCATIONS or CITY_THANAS are imported
      const imports = p1.split(',').map(s => s.trim());
      const keep = imports.filter(i => i === 'LOCATIONS' || i === 'CITY_THANAS');
      if (keep.length > 0) {
        return `import { ${keep.join(', ')} } from '${p2}constants';`;
      }
      return '';
    }}
  ]);
}

// Remove mockData.js
if (fs.existsSync(mockDataPath)) {
  fs.unlinkSync(mockDataPath);
  console.log('Deleted mockData.js');
}

